### PR TITLE
Feat/uri support mac

### PR DESCRIPTION
**Pull request by saledjenic**
closes https://github.com/status-im/status-desktop/issues/3375

Corresponding PRs:
- https://github.com/status-im/nimqml/pull/39
- https://github.com/status-im/dotherside/pull/61

Follow this spec:
- https://github.com/status-im/specs/pull/159/files

### CODE REVIEW

**Review by richard-ramos — APPROVED** [review]

**Review by iurimatias — CHANGES_REQUESTED** [review]
1

**Comment by iurimatias** [comment]
I got this with safari using `status://public/test4`



https://user-images.githubusercontent.com/176720/159934084-2bc17114-d491-40d8-8010-2b2ccfb41636.mov

**Comment by iurimatias** [comment]
@saledjenic see video below, rejoining same channel using deep link makes it unselectable. I cannot replicate this if using the normal join modal
https://user-images.githubusercontent.com/176720/159934988-5e44b9d8-1a40-4e4d-bff6-ded6c30014d1.mov

File: src/app/modules/main/controller.nim [hunk 1]
```diff
@@ -171,8 +171,12 @@ proc init*(self: Controller) =
...
   self.events.on(SIGNAL_MAKE_SECTION_CHAT_ACTIVE) do(e: Args):
     var args = ActiveSectionChatArgs(e)
-    let sectionType = if args.sectionId == conf.CHAT_SECTION_ID: SectionType.Chat else: SectionType.Community
-    self.setActiveSection(args.sectionId, sectionType)
+    self.setActiveSection(args.sectionId)
+
+  self.events.on(SIGNAL_STATUS_URL_REQUESTED) do(e: Args):
+    var args = StatusUrlArgs(e)
+    self.delegate.onStatusUrlRequested(args.action, args.communityId, args.chatId, args.url, args.userId, 
+    args.groupName, args.listOfUserIds)
```
Comment (jrainville) [inline src/app/modules/main/controller.nim:179]: nit: can you add a tab here to separate the different lines

File: src/app/modules/main/module.nim [hunk 2]
```diff
@@ -695,3 +713,57 @@ method newCommunityMembershipRequestReceived*[T](self: Module[T], membershipRequ
...
+    let item = self.view.model().getItemById(conf.CHAT_SECTION_ID)
+    self.setActiveSection(item)
+    self.statusUrlGroupName = groupName
+    # if there are more than 20 members added to the url, we add only first 19
+    self.statusUrlGroupMembersCount = if listOfUserIds.len <= MaxMembersInGroupChatWithoutAdmin: listOfUserIds.len else: MaxMembersInGroupChatWithoutAdmin
+    var i = 0
+    for id in listOfUserIds:
+      if(i >= MaxMembersInGroupChatWithoutAdmin):
+        break
+      i = i + 1
```
Comment (jrainville) [inline src/app/modules/main/module.nim:750]: You can use `i.inc()`

**Review by jrainville — APPROVED** [review]
Code looks good

**Comment by anastasiyaig** [comment]
@saledjenic please rebase, thanks

**Comment by saledjenic** [comment]
> @saledjenic please rebase, thanks

@anastasiyaig rebased again

**Comment by anastasiyaig** [comment]
@saledjenic 

- `/browse/[url]` this pattern opens a browser , but does not open a page, so `status-im://browse/google.com` does not open google.com
- `status-im://u/anastasija@stateofus.eth` is opening some weirdo profile, should be support only exact ENS name like `anastasija` or `anastasija@stateofus.eth` too? 

Screenshot 2022-05-31 at 10 07 16

- `status-im://community-requests/0x0271148720b640eafbe64c9b709c7e105584bb30c2602a55d0473853c4e192c938` does not open a community
- `status-im://cr/0x0271148720b640eafbe64c9b709c7e105584bb30c2602a55d0473853c4e192c938` does not open community
- `status-im://community-channel/1` or `status-im://cc/1` are not opening community with channel `1` and the channel is not shown as active

**Comment by anastasiyaig** [comment]
**QA NOTE:** https://ethstatus.testrail.net/index.php?/runs/view/9309&group_by=tests:assignedto_id&group_order=asc

**Comment by saledjenic** [comment]
@anastasiyaig I've checked issues you stated:

> /browse/[url] this pattern opens a browser , but does not open a page, so status-im://browse/google.com does not open google.com

This works perfectly fine, just be sure that you typed `status-im://browser/google.com`.

> status-im://u/anastasija@stateofus.eth is opening some weirdo profile, should be support only exact ENS name like anastasija or anastasija@stateofus.eth too?

This was a bug, actually we were displaying user's popup even for a public key or ens name which is not the correct one, that's fixed now.

> status-im://community-requests/0x0271148720b640eafbe64c9b709c7e105584bb30c2602a55d0473853c4e192c938 does not open a community
> status-im://cr/0x0271148720b640eafbe64c9b709c7e105584bb30c2602a55d0473853c4e192c938 does not open community

Yes, it doesn't open a community, it just sends a request to join community. If you just want to switch to a community use `c`, `cr` is for request to join community, follow [this](https://github.com/status-im/specs/pull/159/files).

> status-im://community-channel/1 or status-im://cc/1 are not opening community with channel 1 and the channel is not shown as active

Following spec, you should add community channel id, instead of `1`, it is composed of `communityId` + `chatId` that way we can know what exactly to show, otherwise we cannot do that.

**Comment by anastasiyaig** [comment]
@iurimatias how user will know community chat id ? it is not visible in UI , therefore this use case is not actually live

**Comment by anastasiyaig** [comment]
so the discussion about how to get chatId is still ongoing and i dont want to hold the PR because of that (since the logic is implemented anyway) . If there will be a ui ticket to show the chatId -> then it will be done separately and i will retest specific scenarios for community

**Comment by saledjenic** [comment]
@iurimatias this issue is approved and tested, but I cannot merge it since you requested changes [here](https://github.com/status-im/status-desktop/pull/5087#pullrequestreview-920318199), so please either remove requested changes or feel free to merge this PR.

**Review by iurimatias — APPROVED** [review]

**Comment by iurimatias** [comment]
@saledjenic done

**Pull request merged**